src/selenium/trackPage.js

```diff
@@ -0,0 +1,248 @@
+};
+
+// Get the visibility status of the selected sessions after toggling the bookmark button
+TrackPage.starredThenSessionStatus = function() {
+  const self = this;
+  const promiseArr = idArr.map(function(elem) {
+    return self.find(By.id(elem));
+  });
+
+  return self.toggleStarredButton().then(self.getElemsDisplayStatus.bind(null, promiseArr));
+};
+
+// Takes in an array of filters, apply them sequentially and send visibility results of selected sessions on each filter
+TrackPage.filterCombination = function(filtersArr) {
+  const self = this;
+  const filterObjectFunc = {
+    'trackselect': function() {
+      return self.filterThenSessionStatus('true');
+    },
+
+    'trackunselect': function() {
+      return self.filterThenSessionStatus('false');
+    },
+
+    'search': function() {
+      return self.searchThenSessionStatus('wel');
+    },
+
+    'unsearch': function() {
+      return self.searchThenSessionStatus('');
+    },
+
+    'starred': function() {
+      return self.starredThenSessionStatus();
+    },
+
+    'unstarred': function() {
+      return self.starredThenSessionStatus();
+    }
+  };
+
+  const serialPromise = function series(arrayOfPromises) {
+    const results = [];
+
+    return arrayOfPromises.reduce(function(seriesPromise, promise) {
+      return seriesPromise.then(function() {
+        return promise.then(function(result) {
+          results.push(result);
+        });
+      });
+    }, Promise.resolve()).then(function() {
+      return results;
+    });
+  };
+
+  let filtersArrProm = [];
+
+  filtersArrProm = filtersArr.map(function(filter) {
+    return filterObjectFunc[filter]();
+  });
+
+  return serialPromise(filtersArrProm);
+};
+
+TrackPage.checkTrackFilterDirectLink = function() {
+  const self = this;
+  const trackIdArr = [];
+
+  function pushId(trackElem) {
+    trackElem.getAttribute('id').then(function(id) {
+      trackIdArr.push(id);
+    });
+  }
+
+  return new Promise(function(resolve) {
+    self.findAll(By.className('room-filter')).then(function(trackElems) {
+      trackElems.forEach(function(trackElem) {
+        trackElem.isDisplayed().then(function(val) {
+          if (val === true) {
+            pushId(trackElem);
+          }
+        });
+      });
+    }).then(function() {
+      resolve(trackIdArr);
+    });
+  });
+};
+
+TrackPage.checkFilterDynamicLink = function() {
+  const self = this;
+  const promiseArr = [];
+
+  promiseArr.push(self.activeRooms());
+  promiseArr.push(self.activeTracks());
+  return Promise.all(promiseArr);
+};
+
+TrackPage.checkVideo = function() {
+  const self = this;
+
+  return new Promise(function(resolve) {
+    self.find(By.id('title-3015')).then(self.click).then(function() {
+      self.find(By.id('video-3015')).then(function(elem) {
+        elem.getAttribute('src').then(function(url) {
+          resolve(self.isLinkBroken(url));
+        });
+      });
+    });
+  });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must be padded by blank lines. (padded-blocks)](https://app.codacy.com/app/triplez/open-event-webapp/pullRequest?prid=3226522)